lib/taxman/taxman2023/factor.rb

```
@@ -12,7 +12,11 @@ def initialize(**params)
     end
 
     def self.amount(context)
-      new(**context.slice(*params)).amount
+      parameters = params.each_with_object({}) do |param, values|
```

why is this change necessary?

---

There was a bunch of things that were silently failing with similar nil errors to what you were using half a day on the other day. This gives us back the strictness that we had before hand with named parameters as slice does not complain if an element is missing.

So long story short, it makes debugging simpler and fails fast.

When we get around to the 2024 changes, this will make it easier to ensure that we can get them done quickly and not hunt small bugs for ages.

Or said in another way, it was faster for me to implement this than it was to hunt down all the issues we had with not passing in correct parameters.